### PR TITLE
CU-9rrjbq Menu changes preparing for content specific to 1.0

### PR DESCRIPTION
**Pull request by raulraja**
- Removes references to incubator
- Removes links to articles that where mostly encoded in terms of Kinds (to be reviewed in content PRs if content remains relevant)
- Introduces `Training` && `Support` links currently pointing exclusively to 47 Degrees until other companies also want to announce themselves there. /cc @garthgilmour Not sure if Instil is offering Arrow trainings and would like to also be featured here in a dropdown link menu.
- Rearranges menus in core, fx and optics to be consitent.
- Adds link to suspend () -> A in types in Fx pointing to IO removal docs in favor of suspend

### CODE REVIEW

**Comment by franciscodr** [comment]
Task linked: [CU-9rrjbq Polish current documentation to remove obsolete items](https://app.clickup.com/t/9rrjbq)

**Review by franciscodr — APPROVED** [review]

File: docs/_data/sidebar-core.yml [hunk 1]
```diff
@@ -3,139 +3,62 @@ options:
...
 
-      - title: Linting
-        url: /quickstart/linting/
-
-      - title: Libraries
-        url: /quickstart/libraries/
-
       - title: Media
-        url: https://media.arrow-kt.io
+        url: https://media.arrow-kt.io/tags/optics/
```
Comment (nomisRev) [inline docs/_data/sidebar-core.yml]: Why `/tags/optics`?

Comment (raulraja) [reply docs/_data/sidebar-core.yml]: ops, wrong copy/pasta. This was supposed to be in the Media link in the optics site, not core.

Comment (raulraja) [reply docs/_data/sidebar-core.yml]: should be fixed now, each lib pointing to their media tag

File: docs/_data/sidebar-fx.yml [hunk 2]
```diff
@@ -40,27 +36,10 @@ options:
       - title: Semaphore
```
Comment (nomisRev) [inline docs/_data/sidebar-fx.yml]: I cannot add multiline suggestions for code that wasn't modified it seems.

`ConcurrrentVar`, `Promise` & `Semaphore` are deprecated in favor of KotlinX versions of these data types. KotlinX has the following types with the same semantics: `Channel()`, `CompletableDeferred()` and `Semaphore(..)`.

**Review by nomisRev — APPROVED** [review]
Looks good to me. A couple entries for Arrow Fx can be removed due to deprecation. I only reviewed the content for the website. I'm not very familiar with Jekyll or JS.

File: docs/_data/sidebar-fx.yml (hunk 2 above)
Comment (rachelcarmena) [reply docs/_data/sidebar-fx.yml]: Thanks @nomisRev !!

**Pull request merged**